JamGame/Assets/Scripts/Employee/Controller/ControllerImpl.cs

```diff
@@ -127,23 +134,23 @@ private void CorrectMovement()
             }
         }
 
-        public void Teleport(NeedProvider needProvider)
+        public void Teleport(Vector3 destination)
         {
-            transform.position = needProvider.transform.position + (agent.baseOffset * Vector3.up);
+            transform.position = destination + (agent.baseOffset * Vector3.up);
             state = State.BuildingPath;
         }
 
-        public float? ComputePathLength(NeedProvider need_provider)
+        public float? ApproximatePathLength(Vector3 destination)
```

Comment:
Can you return Optional for descriptive return? I don't understand what is null is mean.

Reply:
Discussed